src/main/java/com/library/libraryexercise/controller/LibraryController.java

```diff
@@ -12,48 +13,29 @@
 @RequestMapping("/library")
 public class LibraryController {
 
-    private final List<Book> books = new ArrayList<>();
+    private final LibraryService libraryService;
 
-    public LibraryController() {
-        books.add(new Book("Harry Potter i więzień Azkabanu", "J.K. Rowling", "Fantasy", 300));
+    @Autowired
+    public LibraryController(LibraryService libraryService) {
+        this.libraryService = libraryService;
     }
 
     @GetMapping("/books/{title}")
     public Book getBook(@PathVariable String title, @RequestParam String author) {
-        return books.stream()
-                .filter(book -> book.getTitle().equals(title))
-                .findFirst()
-                .orElseThrow(() -> new BookNotFoundException("Książka o tym tytule: " + title + " nie znajduje się w bazie"));
+        return libraryService.getBook(title, author);
     }
 
     @PutMapping("/books/{title}")
-    public ResponseEntity<Book> addBook(@PathVariable String title,
-                                        @RequestParam String author,
-                                        @RequestParam String genre,
-                                        @RequestParam int pageCount) {
-        boolean exists = books.stream()
-                .anyMatch((book -> book.getTitle().equals(title)));
-
-        if (exists) {
-            return ResponseEntity.badRequest().body(null);
-        }
-
-        Book newBook = new Book(title, author, genre, pageCount);
-
-        books.add(newBook);
-
-        return ResponseEntity.ok(newBook);
+    public Optional<Book> addBook(@PathVariable String title,
+                                  @RequestParam String author,
+                                  @RequestParam String genre,
+                                  @RequestParam int pageCount) {
+        return libraryService.addBook(title, author, genre, pageCount);
```

Comment:
Put should be used for updating or replacing object. So for creation i would use Post and not use multiple @PathVariable and @RequestParam and use new class for example CreateBookRequest with all necessary fields, and pass it to service that will use it to create entity